### PR TITLE
FIX: Histogram module selected value not displayed

### LINKED ISSUE #1295: Selected value of the drop-down is not displayed
**Please Note: We are preparing to participate in Google Code-in, and have reserved this issue for participants in GCI - but we'd love to have your help with another one! Please check out https://code.publiclab.org to see more**

Selected value from the dropdown is not being displayed in the histogram module
Screenshot from 2019-11-12 00-43-42

### PR DESCRIPTION
**Pull request by VladimirMikulic**
Fixes #1295

The histogram module values are "true" and "false".
jQuery would coerce them to Boolean true and false instead of String
"true" and "false" which would result in empty value being displayed.

ezgif com-video-to-gif (4)

@keshav234156 could you review this? Thanks.

### CODE REVIEW

File: examples/lib/defaultHtmlStepUi.js [hunk 1]
```diff
@@ -331,7 +331,7 @@ function DefaultHtmlStepUi(_sequencer, options) {
               .data('initValue', step.options[i]);
           if (inputs[i].type.toLowerCase() === 'select')
             $step('div[name="' + i + '"] select')
-              .val(step.options[i])
+              .val(String(step.options[i]))
```
Comment (HarshKhandeparkar) [inline examples/lib/defaultHtmlStepUi.js]: Just make sure that this does not break any other module.

Comment (VladimirMikulic) [reply examples/lib/defaultHtmlStepUi.js]: No, it does not because everything is a string in the DOM.

Comment (HarshKhandeparkar) [reply examples/lib/defaultHtmlStepUi.js]: Ok great!

**Comment by keshav234156** [comment]
@VladimirMikulic @HarshKhandeparkar cause of the problem seems to be correct. But was just wondering, we also have a "select" type option in Edge-detect.In Edge-detect datatype is identified as a string but in Histogram, data-type is identified as Bool. Do you know why that happening?

**Comment by HarshKhandeparkar** [comment]
I'll have to check. I'll check later.

**Comment by blurry-x-face** [comment]
https://github.com/publiclab/image-sequencer/blob/bfff7e7b999bf256205d412a13f16e154a63beb1/src/modules/Histogram/Module.js#L14

Error is because of this line

**Comment by VladimirMikulic** [comment]
Thanks @blurry-x-face. Yes, that is the error and I've updated the PR accordingly but I'll still keep my String constructor in defaultHtmlStepUi. It will serve like a security mechanism, if a programmer forgets to parse the value as a string, it will do it for him since everything in the DOM is a string.

@keshav234156 this is ready, could you also approve this on GCI dashboard? Thanks.

**Comment by keshav234156** [comment]
@VladimirMikulic Let's protect this with a UI test. Would you like to add a UI test for the test which check that all the module the default options are displayed . I can publish a hard task related to this on the GCI dashboard if you want.

**Comment by VladimirMikulic** [comment]
@keshav234156 I can't promise anything, but if you publish it I'll see what I can do.

**Comment by blurry-x-face** [comment]
@keshav234156  I'll be happy to write those UI tests!

**Comment by keshav234156** [comment]
@blurry-x-face Ok, Go Ahead!!

**Comment by keshav234156** [comment]
@jywarren it's ready for your final review!!!

**Comment by jywarren** [comment]
This is weird. I used to see this:

image

And i still see that on https://github.com/publiclab/image-sequencer/pull/1453

Here I see this:

image

And there's no option to `Update branch` -- do you know why? Can you update it yourself by rebasing? 

Thanks!

**Comment by VladimirMikulic** [comment]
@jywarren now it should be fine. Thanks.

**Comment by jywarren** [comment]
I'm going to try to squash and merge. Maybe it just can tell it doesn't need to re-run checks?

**Comment by VladimirMikulic** [comment]
:+1:

**Pull request merged**